src/handler.ts

```diff
@@ -0,0 +1,685 @@
+        '${game_directory}': config.gameDirectory || resolve(config.root),
+        '${assets_root}': assetPath,
+        '${game_assets}': assetPath,
+        '${version_type}': config.version.type,
+        '${clientid}': config.authorization.client_token || config.authorization.access_token,
+        '${resolution_width}': config.window ? config.window.width : 856,
+        '${resolution_height}': config.window ? config.window.height : 482,
+    };
+
+    if (config.authorization.meta.demo && (config.features ? !config.features.includes('is_demo_user') : true))
+        args.push('--demo');
+
+    if (config.window) {
+        if (config.window.fullscreen) {
+            args.push('--fullscreen');
+        } else {
+            if (config.window.width) args.push('--width', config.window.width.toString());
+            if (config.window.height) args.push('--height', config.window.height.toString());
+        }
+    }
+
+    if (config.quickPlay) {
+        const types = {
+            singleplayer: '--quickPlaySingleplayer',
+            multiplayer: '--quickPlayMultiplayer',
+            realms: '--quickPlayRealms',
+            legacy: null,
+        };
+
+        const { type, identifier, path } = config.quickPlay;
+        const keys = Object.keys(types);
+        if (!keys.includes(type)) {
+            log('debug', `quickPlay type is not valid. Valid types are: ${keys.join(', ')}`);
+        } else {
+            const returnArgs =
+                type === 'legacy'
+                    ? ['--server', identifier.split(':')[0], '--port', identifier.split(':')[1] || '25565']
+                    : [types[type], identifier];
+
+            if (path) returnArgs.push('--quickPlayPath', path);
+            args.concat(returnArgs);
+        }
+    }
+
+    if (config.proxy) {
+        args.push('--proxyHost', config.proxy.host, '--proxyPort', config.proxy.port || '8080');
+
+        if (config.proxy.username) args.push('--proxyUser', config.proxy.username);
+        if (config.proxy.password) args.push('--proxyPass', config.proxy.password);
+    }
+
+    log('debug', 'Set launch options');
+    return args.map((arg) =>
+        Object.entries(fields).reduce(
+            (acc, [placeholder, replacement]) =>
+                acc.replace(new RegExp(placeholder.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'g'), replacement),
+            arg,
+        ),
+    );
+};
+
+const getJVM = async () =>
+    ({
+        windows: '-XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump',
+        osx: '-XstartOnFirstThread',
+        linux: '-Xss1M',
+    })[getOS()];
+
+const getMemory = () => {
+    if (typeof config.memory.min === 'number' && typeof config.memory.max === 'number') {
+        if (config.memory.max < config.memory.min) {
+            log('debug', 'MIN memory is higher then MAX! Resetting!');
+            setConfig('memory', { min: Math.pow(2, 9), max: Math.pow(2, 10) });
```

Comment:
I still don't get why you're doing a global variable, this is not a clean way to keep the state.